packages/core/src/Camera.ts

```diff
@@ -609,10 +609,19 @@ export class Camera extends Component {
   }
 
   private _updatePixelViewport(): void {
-    const viewport = this._viewport;
-    const width = this._renderTarget?.width ?? this.engine.canvas.width;
-    const height = this._renderTarget?.height ?? this.engine.canvas.height;
+    let width: number, height: number;
+
+    const renderTarget = this._renderTarget;
+    if (renderTarget) {
+      width = renderTarget.width;
+      height = renderTarget.height;
+    } else {
+      const canvas = this.engine.canvas;
+      width = canvas.width;
+      height = canvas.height;
+    }
 
+    const viewport = this._viewport;
```

Comment:
The `_lastAspectSize` attribute can be optimized.

Reply:
Done